### PR TITLE
Fix MSI support and add minimal UI

### PR DESCRIPTION
**Pull request by aerkiaga**
This pull request gets MSI support to a state where it can install and uninstall a simple program (under Wine). It also adds a simple setup UI with support for both install and remove actions. It cannot yet upgrade an installation in-place or perform other tasks like setting up shortcuts, modifying environment variables or writing registry values.

Depends on https://github.com/mdsteele/rust-msi/pull/16.

The UI layout is inspired on that of an existing installer, with less dialogs/controls and some modifications: the text has been made larger (size 10, up from 8), and all sentences have been reworded, notably some buttons for clarity ("Next" -> "Install"/"Remove", "Yes" -> "Abort", "No" -> "Continue"). Also, no icons or banners are present; unnecessary though they are, I could draw up some (following any indications) to make the installer more closely follow conventional UIs (edit: done [here](https://github.com/aerkiaga/cargo-bundle/tree/msi-banner), I'll PR if desired).

Edit: I've commented on the fixes. Everything else is not strictly necessary for the installer to work.

### CODE REVIEW

File: src/bundle/msi_bundle.rs [hunk 1]
```diff
@@ -158,6 +171,7 @@ fn set_summary_info(package: &mut Package, package_guid: Uuid, settings: &Settin
     }
     let creating_app = format!("cargo-bundle v{}", crate_version!());
     summary_info.set_creating_application(creating_app);
+    summary_info.set_word_count(2);
```
Comment (aerkiaga) [inline src/bundle/msi_bundle.rs:174]: This fix requires the `rust-msi` patch. Without this line, the installer will attempt to copy files from outside the package.

File: src/bundle/msi_bundle.rs [hunk 2]
```diff
@@ -429,9 +464,9 @@ fn create_feature_table(package: &mut Package, settings: &Settings) -> crate::Re
         msi::Value::from(settings.bundle_name()),
         msi::Value::Null,
         msi::Value::Int(1),
-        msi::Value::Int(3),
+        msi::Value::Int(1),
```
Comment (aerkiaga) [inline src/bundle/msi_bundle.rs:467]: With Feature Level 3 and `INSTALLLEVEL` 1, this feature will never be installed.

File: src/bundle/msi_bundle.rs [hunk 3]
```diff
@@ -591,7 +626,7 @@ fn create_file_table(package: &mut Package, cabinets: &[CabinetInfo]) -> crate::
     for cabinet in cabinets.iter() {
         for resource in cabinet.resources.iter() {
             rows.push(vec![
-                msi::Value::Str(format!("r{sequence:04}")),
+                msi::Value::Str(resource.filename.clone()),
```
Comment (aerkiaga) [inline src/bundle/msi_bundle.rs:629]: This makes the File entry accessible to the Components table.

File: src/bundle/settings.rs [hunk 4]
```diff
@@ -168,6 +168,17 @@ impl Settings {
...
+            Some(x) => match x {
+                PackageType::OsxBundle
+                | PackageType::IosBundle
+                | PackageType::Deb
+                | PackageType::Rpm => "",
+                PackageType::WindowsMsi => ".exe",
+            },
+            None => "",
+        };
+        binary_name += binary_extension;
```
Comment (aerkiaga) [inline src/bundle/settings.rs:181]: Fixes extension handling

**Comment by mdsteele** [comment]
This looks reasonable to me.  I've just published a [v0.6.0](https://crates.io/crates/msi/0.6.0) of the msi crate with the new `set_word_count` method, so you should be set to update `Cargo.toml`.

**Comment by mdsteele** [comment]
Looks like just need to fix some clippy errors.

**Comment by mdsteele** [comment]
Thanks!

**Pull request merged**